h/search/query.py

```diff
@@ -271,6 +271,14 @@ def __call__(self, search, _):
         return search.exclude("exists", field="deleted")
 
 
+class HiddenFilter(object):
+
+    """A filter that only returns documents which are not hidden."""
+
+    def __call__(self, search, _):
+        return search.filter("term", hidden=False)
```

Comment:
When you write tests for this can you check the interaction between the NipsaFilter and this HiddenFilter. I suspect they won't work correctly together because logic outside of each filter is an "AND" rather than an "OR" in which case we might need to combine some of the Nipsa logic into this filter (basically if the user is the author of the hidden annotation don't hide it).